packages/core/src/controllers/relayer.ts

```diff
@@ -181,8 +187,19 @@ export class Relayer extends IRelayer {
         resolvePromise = resolve;
         this.subscriber.on(SUBSCRIBER_EVENTS.created, onSubCreated);
       }),
-      new Promise<void>(async (resolve) => {
-        const result = await this.subscriber.subscribe(topic, opts);
+      new Promise<void>(async (resolve, reject) => {
+        const result = await this.subscriber
+          .subscribe(topic, {
+            internal: {
+              throwOnFailedPublish: shouldThrowOnFailure,
+            },
+            ...opts,
```

Comment:
Wouldn't `opts` override `internal.throwOnFailedPublish` ?
```suggestion
            ...opts,
            internal: {
              ...opts.internal,
              throwOnFailedPublish: shouldThrowOnFailure,
            },
```

Reply:
its intended to be overriden if its defined